### PR TITLE
Use txzmq in backend (closes #6360).

### PR DESCRIPTION
**Pull request by andrejb**

### CODE REVIEW

**Comment by ivanalejandro0** [comment]
This needs to be carefully rebased after changes made on https://github.com/leapcode/bitmask_client/pull/826

**Comment by ivanalejandro0** [comment]
@andrejb could you rebase&push this?

here are the changes for the rebase:

```diff
diff --git a/src/leap/bitmask/backend/backend_proxy.py b/src/leap/bitmask/backend/backend_proxy.py
index 9e62372..09cb6ef 100644
--- a/src/leap/bitmask/backend/backend_proxy.py
+++ b/src/leap/bitmask/backend/backend_proxy.py
@@ -58,17 +58,13 @@ class ZmqREQConnection(threading.Thread):
         self._on_recv = on_recv
         self._stream = None
 
-<<<<<<< HEAD
         self.settings = Settings()
 
-        # initialize ZMQ stuff:
-=======
     def _init_zmq(self):
         """
         Configure the zmq components and connection.
         """
         logger.debug("Setting up ZMQ connection to server...")
->>>>>>> Use txzmq in backend (closes #6360).
         context = zmq.Context()
         socket = context.socket(zmq.REQ)
 
diff --git a/src/leap/bitmask/backend/components.py b/src/leap/bitmask/backend/components.py
index a5bd42c..0a2f102 100644
--- a/src/leap/bitmask/backend/components.py
+++ b/src/leap/bitmask/backend/components.py
@@ -63,12 +63,8 @@ from leap.common import certs as leap_certs
 
 from leap.keymanager import openpgp
 
-<<<<<<< HEAD
-from leap.soledad.client.secrets import NoStorageSecret, PassphraseTooShort
-=======
 from leap.soledad.client.secrets import PassphraseTooShort
 from leap.soledad.client.secrets import NoStorageSecret
->>>>>>> Use txzmq in backend (closes #6360).
 
 logger = logging.getLogger(__name__)
```

File: src/leap/bitmask/backend/backend_proxy.py [hunk 1]
```diff
@@ -157,6 +225,8 @@ def _api_call(self, *args, **kwargs):
             'arguments': kwargs,
         }
 
+        request_json = None
```
Comment (ivanalejandro0) [inline src/leap/bitmask/backend/backend_proxy.py]: This is not needed I think.

**Comment by ivanalejandro0** [comment]
We should add `txzmq` on the requirements file.

File: src/leap/bitmask/backend/backend_proxy.py [hunk 2]
```diff
@@ -36,33 +38,40 @@
...
+        :param on_recv: The callback to be executed when a message is
+            received.
+        :type on_recv: callable(msg)
+        """
+        threading.Thread.__init__(self)
+        self._server_address = server_address
+        self._on_recv = on_recv
+        self._stream = None
 
         self.settings = Settings()
```
Comment (ivanalejandro0) [inline src/leap/bitmask/backend/backend_proxy.py]: This line belongs to the `BackendProxy.__init__` method.

File: src/leap/bitmask/backend/backend_proxy.py [hunk 3]
```diff
@@ -36,33 +38,39 @@
...
-
-    def __init__(self):
-        generate_zmq_certificates_if_needed()
-
-        self._socket = None
+    def __init__(self, server_address, on_recv):
+        """
+        Initialize the connection.
 
-        self.settings = Settings()
```
Comment (ivanalejandro0) [inline src/leap/bitmask/backend/backend_proxy.py]: This line is needed here. See https://github.com/leapcode/bitmask_client/blob/develop/src/leap/bitmask/gui/mainwindow.py#L129

**Comment by ivanalejandro0** [comment]
@andrejb I've rebased this against the latest `develop` and pushed results to https://github.com/ivanalejandro0/bitmask_client/tree/feature/6360_use-txzmq-in-backend
Please use that, see the comments in here and update this pullreq.

File: src/leap/bitmask/backend/backend_proxy.py (hunk 1 above)
Comment (andrejb) [inline src/leap/bitmask/backend/backend_proxy.py]: Not needed indeed, but maybe good practice to ensure the variable exists in enclosing context? Couldn't find a reference, though.

File: src/leap/bitmask/backend/backend_proxy.py (hunk 3 above)
Comment (ivanalejandro0) [inline src/leap/bitmask/backend/backend_proxy.py]: my bad... this goes into the `BackendProxy` class, and already is there.

**Comment by ivanalejandro0** [comment]
The only thing we have pending that holds on this commit to be merged is the "gazillion of logs in the terminal" issue. It seems that having `coloredlogs` installed it does not happen.
So, we can figure out why this is happenning or just add `coloredlogs` as a dependency.

**Comment by kalikaneko** [comment]
if I reember correctly, coloredlogs also makes some logs not show.
I think it might have to do with installing custom handlers: we'd be stepping on each other's.

**Comment by kalikaneko** [comment]
I think I would use coloredlogs as a dep, then merge your work with the zmq handler, and later investigate if the problem is still happening. what do you think?

**Comment by ivanalejandro0** [comment]
It works fine for me.
@andrejb please add `coloredlogs` to the `pkg/requirements.pip` file and rebase this to develop.

**Pull request merged**